src/authentication/background/auth-background.ts

```diff
@@ -0,0 +1,49 @@
+import { AuthService } from 'src/authentication/background/auth-service'
+import {
+    FirebaseFunctionsAuth,
+    FirebaseFunctionsSubscription,
+} from 'src/authentication/background/firebase-functions-subscription'
+import { remoteEventEmitter } from 'src/util/webextensionRPC'
+import { AuthFirebase } from 'src/authentication/background/auth-firebase'
+import { MockAuthImplementation } from 'src/authentication/background/mocks/auth-mocks'
+
+export class AuthBackground {
+    authService: AuthService
+    subscriptionServerFunctions: FirebaseFunctionsSubscription
+    authServerFunctions: FirebaseFunctionsAuth
+
+    constructor(
+        authService: AuthService = null,
+        subscriptionServerFunctions: FirebaseFunctionsSubscription = null,
+        authServerFunctions: FirebaseFunctionsAuth = null,
+    ) {
+        if (authService != null) {
+            this.authService = authService
+        } else {
```

Comment:
Here I'd say that setting up a developer implementation is the responsibility of whoever creates this object. This class doesn't need to know anything itself. As I did with the Sync, you can pass these into the `createBackgroundModules` function, so everything related to setting up a developer environment is handled in `background.ts` (which can without any problem import a helper function from inside this module to create things.)